app/experimenter/nimbus-ui/src/components/AppLayoutWithSidebar/index.tsx

```diff
@@ -106,9 +110,20 @@ export const AppLayoutWithSidebar = ({
                     </LinkNav>
                   ) : (
                     <DisabledItem name="Results" testId="show-no-results">
-                      {status?.accepted
-                        ? "Waiting for experiment to launch"
-                        : "Experiment results not yet ready"}
+                      {status?.accepted ? (
+                        "Waiting for experiment to launch"
+                      ) : analysisError ? (
```

Comment:
I am normally not a fan of nested ternaries, but the logic seemed clear enough here.

Reply:
@pdehaan Thanks for your comment!

I dug a bit deeper into what `analysisError` _is_ - unfortunately it's a generic "the thing didn't work" catch-all. It's the same `SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data` message if I force a `500` internal server error error as well as if the API call `404`s because we attempt to JSON parse the response from the server no matter what.

The package we're using to fetch the data in a React hook (and receive `analysisError` from), `react-async-hook`, [doesn't actually hook (pun intended) into HTTP here](https://github.com/slorber/react-async-hook/issues/20#issuecomment-574543580), so I can't parse the error code/status from `analysisError`.

If we want to display the message from the server or even display the error code to help see what's going on, we'll need a little refactoring. I actually spent half an hour attempting it, as I thought I could throw a `try`/`catch` into our `fetchData` function (supplied to the aforementioned package) to use _that_ caught error instead which should contain the error code and message, but couldn't get it working like I wanted.

I'm hoping that with the bug fix included, this scenario won't occur very often at all. When it does, anything going awry on the server-side should already be sent off to Sentry as well as anything that creeps up into our client-side `AppErrorBoundary`.

Since the bug fix here is highest priority and this other stuff is extra anyway, let's make a follow up ticket to see if we can capture the actual error from the HTTP request with code & message?

With that said, I'm unsure if data science is who should be contacted if this errors. Could be a bug on their side, but also on ours. 🤷‍♀️

Reply:
Ana confirmed the text here (it's a shortened version of what it was) - @pdehaan if you think it would be valuable to display the error message from the server instead, we can file a follow up, but otherwise I'll leave it as-is and we can see how often this actually occurs when it's in the wild.